app_widgetbook/lib/main.dart

```diff
@@ -307,9 +307,12 @@ class WidgetbookApp extends StatelessWidget {
                   useCases: [
                     WidgetbookUseCase(
                       name: 'Default',
-                      builder: (context) => AppIconButton(
+                      builder: (context) => AppIconButton.builder(
                         onPressed: () {},
-                        icon: const Icon(Icons.home_filled),
+                        iconBuilder: (context, color, size) => AppIcon.heart(
+                          color: color,
+                          size: size,
+                        ),
```

Comment:
For a button, this looks a bit complicated. Can you explain why you picked this approach over the one that already existed?

I assume this has something to do with displaying the active state of a button.

Reply:
Hmm this is an interesting problem since the icon can change its color. I also had a bit of a hard time understanding the difference between the constructor and named constructor. I think the `itemBuilder` makes a lot of sense but right now it looks like that `iconBuilder` and `icon` aim to do similar things. However, `icon` is for buttons that do not have colored icons that change their color `onHover` and `iconBuilder` is for icons that change color `onHover`. Is this assessment correct?

Reply:
Yes your assessment is correct. However, we can simplify this by using `AppIcons.iconPath` as you see in the latest commit c66dc9a83dc1fabb4289ba8992a7be26ad0dcca8. Now we only need the `iconBuilder` if we want full control over the child icon `Widget` which is needed in the `LikeIconButton` to control the animation.

Reply:
Using a path for icons feels odd to me. But that's because we are using the `.png` files for icons instead of `.svg` or an icon library. What we are doing here is basically building the icon library ourselves and are now implementing the Twitter buttons. I now see two downsides with the `.png` implementation: scaling and working with paths. Do you think it's worth the trouble? Will there be additional problems coming up?

Reply:
I am looking into alternative solutions that involve creating a font from an icon set. This site seems more promising 
https://icomoon.io/app/#/select
It was able to read our icons correctly and generate a font file.
Should I proceed with this?